config/config.go

```diff
@@ -276,26 +330,18 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 			return fmt.Errorf("invalid rule file path %q", rf)
 		}
 	}
+
+	for _, sf := range c.ScrapeConfigFiles {
+		if !patRulePath.MatchString(sf) {
+			return fmt.Errorf("invalid scrape config file path %q", sf)
+		}
+	}
+
 	// Do global overrides and validate unique names.
 	jobNames := map[string]struct{}{}
 	for _, scfg := range c.ScrapeConfigs {
```

Comment:
should we delete this section? is't duplicate with [Line 266](https://github.com/prometheus/prometheus/pull/12019/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R266) ?

Reply:
It is not a duplicate, it is still here to fail faster.

Reply:
if just used for fail faster , the [line 267](https://github.com/prometheus/prometheus/pull/12019/files/eb2aa5845e41d7ebecb9e88b5496b50792dddb0f#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R267)  Validate is still need?  😂 because it has Validated once.

Reply:
I put it here as I don't know if third parties could use getscrapeconfigs without unmarshalling properly. Maybe I can comment the 267 with a comment saying this is checked during unmarshalling.

Reply:
it is also fine to do a comment for this two places which can solve our confused why they validate twice.